src/com/google/enterprise/adaptor/filenet/DocumentTraverser.java

```diff
@@ -213,20 +212,23 @@ private void processDocument(Id guid, DocId docId, Document document,
     String vsDocId = document.get_VersionSeries().get_Id().toString();
     logger.log(Level.FINE, "VersionSeriesID for document is: {0}", vsDocId);
 
-    ActiveMarkingList activeMarkings = document.get_ActiveMarkings();
-    Permissions.Acl permissions =
-        new Permissions(document.get_Permissions(), document.get_Owner())
-        .getAcl();
-    response.setAcl(getAcl(docId, permissions));
-    processInheritedPermissions(docId, activeMarkings, permissions, response);
+    if (options.markAllDocsAsPublic()) {
+      response.setSecure(false);
+    } else {
+      ActiveMarkingList activeMarkings = document.get_ActiveMarkings();
```

Comment:
Is there any merit to passing document down and getting the active markings and permissions in processPermissions? We could do the same for content eventually, and maybe even move what's left here back up into getDocContent.